src/Alert/index.jsx

```diff
@@ -1,37 +1,106 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 import Alert from 'react-bootstrap/Alert';
+import { useMediaQuery } from 'react-responsive';
+import useToggle from '../hooks/useToggle';
 import { Icon } from '..';
+import AlertButtons from './AlertButtons';
+import { alertButtonProps } from './propTypes';
+import { breakpoints } from '../Responsive';
 
 const WrappedAlert = React.forwardRef(({
   children,
   icon,
+  button,
+  dismissible,
+  onClose,
+  closeLabel,
+  show,
+  stacked,
   ...props
-}, ref) => (
-  <Alert
-    {...props}
-    className={`alert-content ${props.className}`}
-    ref={ref}
-  >
-    {icon && <Icon src={icon} className="alert-icon" />}
-    <div className="alert-message-content">
-      {children}
-    </div>
-  </Alert>
-));
+}, ref) => {
+  const [isOpen, /* open */, close] = useToggle(show);
+  const [isStacked, setIsStacked] = useState(stacked);
+  const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth });
+
+  useEffect(() => {
+    if (isExtraSmall) {
+      setIsStacked(true);
+    } else {
+      setIsStacked(stacked);
+    }
+  }, [isExtraSmall, stacked]);
+
+  const handleClose = useCallback((e) => {
+    close();
+    onClose(e);
+  }, []);
+
+  return (
+    <Alert
+      {...props}
+      show={isOpen}
+      className={classNames('alert-content', props.className)}
+      ref={ref}
+    >
+      {icon && <Icon src={icon} className="alert-icon" />}
+      <div
+        className={classNames(
+          'pgn__alert-content-wrapper',
+          {
+            'd-flex align-content-center align-items-center': (dismissible || button) && !isStacked,
+          },
+        )}
+      >
+        <div className="alert-message-content">
+          {children}
+        </div>
+        {(dismissible || button) && (
+          <AlertButtons
+            className={classNames({
+              'mt-3': isStacked,
+              'ml-4': !isStacked,
+            })}
+            button={button}
```

Comment:
It's possible that we may want to offer two actions on an alert (not including dismiss). What do you think about changing `button` to `actions` as an array of `node`s?

Reply:
Having an `actions` type of prop crossed my mind but didn’t implement it that way since the Figma spec didn't explicitly showcase that as a use case. But I agree it may be worth supporting.

The caveat is what happens when there’s 2+ actions and a dismiss button? There's not much real estate for more than 2 buttons here with the right-aligned button variant. We could use an overflow for the buttons like we have started to do elsewhere but I think my suggestion would be to defer on that until there's an explicit use case for it.

I'm for the `actions` prop assuming we have a custom prop type that throws a prop type warning/error when:
* `actions` has length greater than 2
* `actions` has length greater than 1 and `dismissible` is true.